nixarr/monitoring/default.nix

```diff
@@ -0,0 +1,263 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+with lib; let
+  cfg = config.nixarr;
+  
+  # Helper function to determine if a specific exporter should be enabled
+  shouldEnableExporter = service: 
+    cfg.${service}.enable && 
+    (cfg.monitoring.exporters.${service} or true);
+    
+  # Helper function to determine if a service is VPN-confined
+  isVpnConfined = service: cfg.${service}.enable && cfg.${service}.vpn.enable;
+    
+  # Helper function to create a script that extracts API key from config.xml
+  extractApiKeys = pkgs.writeShellApplication {
+    name = "extract-monitoring-api-keys";
+    runtimeInputs = with pkgs; [ dasel coreutils ];
+    text = ''
+      # Create state directory for API keys
+      STATE_DIR="/var/lib/exportarr"
+      mkdir -p "$STATE_DIR"
+      chmod 755 "$STATE_DIR"
+
+      # Function to wait for config file and extract API key
+      wait_and_extract() {
```

Comment:
Why wait/sleep at all?

Reply:
The reason is that it may take a little until the config files show up, so we gotta synchronize it. I tried various approachs (eg. using `ConditionPathExist`, Timers, etc.) but none of them worked well. `ConditionPathExist` is only checked once. Timers work but then we also need to introduce `Restart=always` on all depending units. This is still kind of nasty but I did not find another way, if you have an idea, let me know.

Reply:
Sounds very reasonable. The current solution is "nasty", but only enough to question it. I think it's fine if there's no obvious alternative. Nice work.